src/common/board.rs

```diff
@@ -56,14 +70,28 @@ pub trait Board: Status {
     fn get_digital_interrupt_value(&self, _pin: i32) -> anyhow::Result<u32> {
         anyhow::bail!("this board does not support digital interrupts")
     }
+    /// Get the pin's given duty cycle
     fn get_pwm_duty(&self, pin: i32) -> f64;
+    /// Set the pin to the given duty cycle 
     fn set_pwm_duty(&mut self, pin: i32, duty_cycle_pct: f64) -> anyhow::Result<()>;
+    /// Get the PWM frequency of the pin
     fn get_pwm_frequency(&self, pin: i32) -> anyhow::Result<u64>;
+    /// Set the pin to the given PWM frequency (in Hz). 
+    /// When frequency is 0, it will use the board’s default PWM frequency.
     fn set_pwm_frequency(&mut self, pin: i32, frequency_hz: u64) -> anyhow::Result<()>;
 }
 
+/// An alias for a thread-safe handle to a struct that implements the [Board] trait
 pub type BoardType = Arc<Mutex<dyn Board>>;
 
+/// A test implementation of a generic compute board
+pub struct FakeBoard {
+    analogs: Vec<Rc<RefCell<dyn AnalogReader<u16, Error = anyhow::Error>>>>,
+    i2cs: HashMap<String, Arc<Mutex<FakeI2CHandle>>>,
```

Comment:
we... maybe be getting rid of i2c altogether and having the sensors handle them. so I'd hold off on that  pr to go in before doing too much there - the i2c call is not part of our api.

Reply:
Ok. I don't think I touched anything specifically with i2c though, this change was just me moving `FakeBoard`'s struct definition below the `Board` trait definition, to group related things closer together.

Reply:
Ahh I think I see what you're saying. The `Board` trait itself defines some functions for i2c that aren't part of the Board API. Yeah I was only putting docstrings on methods that I found equivalents for in the Python SDK, which I believe does accurately reflect the api.

Reply:
sort of - here's the ground truth for the [board API](https://github.com/viamrobotics/api/blob/main/proto/viam/component/board/v1/board.proto), python has a lot of client side-only helper functions, which might be going away.

Reply:
Since it is just code motion though I don't think this PR should be blocked on any i2c work. Is that ok with you @randhid ?

Reply:
totally okay.